cli/src/main/java/denominator/cli/Denominator.java

```diff
@@ -309,6 +323,43 @@ public void run() {
       }
     }
 
+    private Function<Object, Object> attemptToTransformCredentials = new Function<Object, Object>() {
+      @Nullable
+      @Override
+      public Object apply(@Nullable Object input) {
+        input = attemptExtractX509Certificate(input);
+        return attemptExtractPrivateKey(input);
+      }
+    };
+
+    private Object attemptExtractX509Certificate(Object value) {
+      if (value != null && value instanceof String && value.toString().contains("BEGIN CERTIFICATE")) {
+        try {
+          X509CertParser x509CertParser = new X509CertParser();
+          x509CertParser.engineInit(new ByteArrayInputStream(value.toString().getBytes()));
+          return x509CertParser.engineRead();
+        } catch (Exception ex) {
+          return value;
+        }
+      }
+      return value;
+    }
+
+    private Object attemptExtractPrivateKey(Object value) {
+      if (value != null && value instanceof String && value.toString().contains("BEGIN RSA PRIVATE KEY")) {
+        try {
+          final PEMKeyPair pemKeyPair = (PEMKeyPair) new PEMParser(new StringReader(value.toString())).readObject();
```

Comment:
generally, don't put final on locals unless they are needed. It is a distracting keyword to folks like me who see it and wonder.. why is this final?